docs/src/ComboBox.doc.js

```diff
@@ -0,0 +1,732 @@
+   "DIY": [
+    'DIY Projects',
+    'DIY Art',
+    'DIY Home decor',
+    'DIY Furniture',
+    'DIY Gifts',
+    'DIY Wall decor',
+    'DIY Clothes',
+    'DIY Christmas decorations',
+    'DIY Christmas gifts',
+    'DIY Wall art'].map((pronoun, index) => ({ label: pronoun, value: 'value'+index }))
+  };
+
+  const [currentCategory, setCurrentCategory] = React.useState("BEAUTY");
+
+  const [suggestedOptions, setSuggestedOptions] = React.useState(CATEGORIES[currentCategory]);
+
+  const [inputValue, setInputValue] = React.useState("");
+
+  const [selectedOption, setSelectedOption] = React.useState();
+
+  const resetOptions = () => {
+    setSuggestedOptions(CATEGORIES[currentCategory]);
+  }
+
+  const handleOnChange = ({ value }) => {
+    setSelectedOption()
+    if (value) {
+      setInputValue(value);
+      const filteredOptions = CATEGORIES[currentCategory].filter((item) =>
+        item.label.toLowerCase().includes(value.toLowerCase()),
+      );
+      setSuggestedOptions(filteredOptions);
+    } else {
+      setInputValue(value);
+      resetOptions();
+    }
+  };
+
+  const handleSelect = ({ item }) => {
+    setInputValue(item.label);
+    setSelectedOption(item);
+    resetOptions();
+  };
+
+  const handleOnBlur = () => {
+    if (!selectedOption) setInputValue("");
+    resetOptions();
+  }
+
+  const handleOnClear = () => {
+    setInputValue("");
+    setSelectedOption()
+    resetOptions();
+  }
+
+  return (
+    <Flex direction="column" gap={10}>
+      <Button
+        onClick={() => {
+          const nextCategory = currentCategory === 'BEAUTY' ? 'DIY' : 'BEAUTY';
+          setCurrentCategory(nextCategory)
+          setSuggestedOptions(CATEGORIES[nextCategory])
+          setInputValue("")
+        }}
+        text={"Change options to " + (currentCategory === "BEAUTY" ? "DIY" : "BEAUTY") + " category"}
+        inline
+        />
+      <ComboBox
+        accessibilityClearButtonLabel="Clear the current value"
+        accessibilityShowButtonLabel="Show a list of categories for this field"
+        id="programaticallySet"
+        inputValue={inputValue}
+        noResultText="No results for your selection"
+        options={suggestedOptions}
+        label="Pin category"
+        size="lg"
+        onBlur={handleOnBlur}
+        onClear={handleOnClear}
+        placeholder="Select a category"
+        onChange={handleOnChange}
+        onSelect={handleSelect}
+        selectedOption={selectedOption}
+      />
+    </Flex>
+  );
+}`}
+      />
+    </MainSection.Subsection>
+    <MainSection.Subsection
+      description={`
+    Include [Tag](/Tag) elements in the input using the \`tags\` prop.
+
+    Note that the \`ComboBox\` component doesn't internally manage tags; therefore, it must be a [controlled component](#Controlled-vs-Uncontrolled). A controlled ComboBox with requires three value props: \`options\`,  \`inputValue\`,  and \`tags\`.
```

Comment:
"A controlled ComboBox requires three value props: \`options\`,  \`inputValue\`,  and \`tags\`."

Reply:
DONE